### PR TITLE
Health check for SSL/TLS services

### PR DESCRIPTION
**Pull request by Carlotronics**
- Add SSL service health check;
- Add timeout feature for StartTLS checks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -963,6 +964,24 @@ services:
...
+        timeout: 5s
+    conditions:
+      - "[CONNECTED] == true"
+      - "[CERTIFICATE_EXPIRATION] > 48h"
+```
+
+
+### Monitoring a service using SSL
+If you have an email server that you want to ensure there are no problems with, monitoring it through SSL
+will serve as a good initial indicator:
```
Comment (TwinProduction) [inline README.md:977]: Any chance you could update the description? It's pretty much the same as the description for `STARTTLS`.

Also, is it really accurate to make the protocol `ssl://` when it's in fact `tls`?

File: client/client.go [hunk 2]
```diff
@@ -57,6 +61,19 @@ func CanPerformStartTLS(address string, config *Config) (connected bool, certifi
...
+// CanPerformSSL checks whether a connection can be established to an address using the SSL protocol
+func CanPerformSSL(address string, config *Config) (connected bool, certificate *x509.Certificate, err error) {
+	conn, err := tls.DialWithDialer(&net.Dialer{Timeout: config.Timeout}, "tcp", address, nil)
+	if err != nil {
+	    return
+	}
+	defer conn.Close()
+
+	chain := conn.ConnectionState().VerifiedChains[0] // VerifiedChains[0] == PeerCertificates[0]
+	certificate = chain[0]
```
Comment (TwinProduction) [inline client/client.go:73]: Would it be possible to check the length before calling these?

I'd rather avoid risking a `panic`.

**Review by TwinProduction — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Carlotronics) [reply README.md]: > Also, is it really accurate to make the protocol ssl:// when it's in fact tls?

You're right, `tls://` would be more accurate. I've updated it.

File: client/client.go (hunk 2 above)
Comment (Carlotronics) [reply client/client.go]: Sure. I've added both checks in c55d680.

**Comment by Carlotronics** [comment]
@TwinProduction please tell me when you'll be ready to merge changes so I can squash last commits. I think it is clearer to keep them distinct for now but they won't be useful anymore in main tree.

**Comment by TwinProduction** [comment]
@Carlotronics Sorry about that, looks like the test failed because of a feature I recently added.

I just fixed it in 428e415; could you pull the changes from master?

**Review by TwinProduction — APPROVED** [review]

**Comment by TwinProduction** [comment]
Thank you for your contribution!

**Comment by TwinProduction** [comment]
Released in [v3.2.0](https://github.com/TwinProduction/gatus/releases/tag/v3.2.0)

**Pull request merged**